lib/auth/native/native.go

```diff
@@ -42,16 +52,46 @@ type keyPair struct {
 type nauth struct {
 	generatedKeysC chan keyPair
 	closeC         chan bool
-	closed         int32
+	mutex          sync.Mutex
+	// in production set to nil (used for tests)
+	testKey *keyPair
 }
 
+// New returns a pointer to a key generator for production purposes
 func New() *nauth {
-	n := nauth{
-		generatedKeysC: make(chan keyPair, PrecalculatedKeysNum),
-		closeC:         make(chan bool),
+	singleton.mutex.Lock()
+	defer singleton.mutex.Unlock()
+
+	if singleton.generatedKeysC == nil && PrecalculatedKeysNum > 0 {
+		singleton.generatedKeysC = make(chan keyPair, PrecalculatedKeysNum)
+		go singleton.precalculateKeys()
+	}
+	return &singleton
+}
+
+// NewTestInstance
+func SetTestKeys() {
```

Comment:
right, but in this implementation test key fixutre is compiled into the production release. I'd prefer to use testauthority as described above to avoid this and for unifs and for the sake of unified usage of testauthority. With your change, we have both testauthority and this new test key injection